Solution to Find Current Time

tavisca-rpawar opened the pull request:
@tavisca-dagrawal Please review my program

tavisca-rpawar committed:
Solution to Find Current Time 

@tavisca-dagrawal Please review my program

tavisca-rpawar committed:
Second Commit

divastavisca commented on Tavisca.Bootcamp.LanguageBasics.Exercise2/Program.cs:
```diff
@@ -13,18 +13,78 @@ static void Main(string[] args)
+
+                    HH.ToString();
+                    MM.ToString();
+                    SS.ToString();
```
I think this has been mistakenly placed here. 
But if not, ToString() does not modify the data type of the variable(C# is strongly typed), but returns a string. ToString() function has a base implementation(defined in object class) and can be overriden in any implmentation

divastavisca commented on Tavisca.Bootcamp.LanguageBasics.Exercise2/Program.cs:
```diff
@@ -13,18 +13,78 @@ static void Main(string[] args)
+                    else if(SS.Equals(0))
+                        result = HH + ":" + MM + ":" + "00";
+                    else
+                        result = HH + ":" + MM + ":" + SS;
```
Instead of using '+' for concatinating strings, try using string.Format.
like this: 
string.Format("{0}:{1}:{2}",hh,mm,ss);
```suggestion
                        result = string.Format("{0}:{1}:{2}",HH,MM,SS);
```
 or try StringBuilder class (Although in here string.Format() is much better )
refer: [String Builder doc by microsoft](https://docs.microsoft.com/en-us/dotnet/api/system.text.stringbuilder?view=netframework-4.8
)
These are more readable.

divastavisca commented on Tavisca.Bootcamp.LanguageBasics.Exercise2/Program.cs:
```diff
@@ -13,18 +13,78 @@ static void Main(string[] args)
+
+                }
+                var exactHour = exactPostTime[i].Split(":")[0];
+                var HH = int.Parse(exactHour);
```
private identifier names should be in camelCase with a leading underscore
For more info on naming coventions, good to read about [Naming conventions](https://github.com/ktaranov/naming-convention/blob/master/C%23%20Coding%20Standards%20and%20Naming%20Conventions.md)

These are standards, and are present to improve code readability

divastavisca commented on Tavisca.Bootcamp.LanguageBasics.Exercise2/Program.cs:
```diff
@@ -13,18 +13,78 @@ static void Main(string[] args)
+                var HH = int.Parse(exactHour);
+                var exactMinute = exactPostTime[i].Split(":")[1];         
+                var MM = int.Parse(exactMinute);
+                var exactSecond = exactPostTime[i].Split(":")[2];
```
string.Split() multiple use, please optimize

divastavisca commented on Tavisca.Bootcamp.LanguageBasics.Exercise2/Program.cs:
```diff
@@ -13,18 +13,78 @@ static void Main(string[] args)
+                    }
+                }
+                // if showPostTime contain minutes then the total number of minutes ago is added to MM which shows current minute 
+                if(showPostTime[i].Contains("minutes"))
```
Your approach is sound, trying to build a time component from a input
like hour from "hours ago"
min from "mins ago"

But there could be have been multiple "hours ago" or "mins ago" and for each element you are finding "result" and overriding it with the next same "hours ago" or "mins ago", so at the termination of your program you may not have a lexicographically smaller result, but the "result" which was calculated from the last input

**Your approach need to be relative in inputs**

And also, consider the scenerio, when input only contains multiple "a few seconds ago".
That is, {"11:11:13","11:11:19","11:11:43"} & {"few seconds ago","few seconds ago".few seconds ago"}

One more suggestion, you could have also used [DateTime](https://docs.microsoft.com/en-us/dotnet/api/system.datetime?view=netframework-4.8)  functions defined in System namespace, they provide you a hell lot of functions which work on date and time.

divastavisca reviewed (changes requested):
Please find my comments.
You need to re evaluate your algorithm.
And try to optimize your code by using .net defined functions. 
Try using intelli-sense of Visual Studio to lookup for the functions which can save your effort of writing very basic things.

tavisca-rpawar committed:
Updated  Solution using DateTime

Yes sir i agree, previous committed code is very basic.
I implemented this code few days after that commit, but due to some Test case failure and i am not able to  commit this one,  now i have resolved the errors.
